internal/engine/handler.go

```diff
@@ -142,6 +175,11 @@ func (e *ExecutorEventHandler) HandleEntityEvent(msg *message.Message) error {
 		if err := e.evt.Publish(events.TopicQueueEntityFlush, msg); err != nil {
 			logger.Err(err).Msg("error publishing flush event")
 		}
+		e.lock.Lock()
+		defer e.lock.Unlock()
+		e.cancels = slices.DeleteFunc(e.cancels, func(cf *context.CancelFunc) bool {
+			return cf == &shutdownCancel
+		})
```

Comment:
Since we're appending a new cancelation function on every call to the handler at line 110, should we call this logic to drop `shutdownCancel` from the `e.cancels` slice before the return on line 171 too?

Reply:
Awesome, thank you :)